File: Stack/stack.c
#include<stdio.h>
#include<stdlib.h>

typedef struct{

	int size;
	int top;
	int *S;

}stack;

void Create(stack *s1){
	
	printf("Enter the size of the stack : ");
	scanf("%d",&s1->size);
	s1->top = -1;
	s1->S = (int *)malloc(s1->size*sizeof(int));

}

void Display(stack st){
	int i;
	for(i=st.top;i>=0;i--){
		printf("%d\n",st.S[i]);
	}
	//printf("\n");
	
}

//the push funtion doesnot return any value
void push(stack *s1,int x){
	
	if(s1->top >= s1->size-1){
		printf("Stack Overflow\n");
	}else{
		s1->top+=1;
		s1->S[s1->top] = x;
	}
	
}

int pop(stack st){
	
	int x = -1;
	if(st.top<=-1){
		printf("Stack Underflow\n");
	}else{
		x = st.S[st.top];
		st.top--;
	}
	
	return x;
	

}

int main(){

	stack s1;
	Create(&s1);
	push(&s1,10);
	push(&s1,20);
	push(&s1,30);
	
	Display(s1);
	


	return 0;
}

File: Stack/stack.java
class Stack{
    int top;
    int size;
    int[]  a;
    
    Stack(int size){
        top = -1;
        this.size = size;
        a = new int[this.size];
    }

    

    public void push(int n){
        if(top >= this.size-1){
            System.out.println("Stack Overflow");
        }else{
            top = top+1;
            a[top] = n;
        }
    }


    public int pop(){
        int num = -1;
        if(top == -1){
            System.out.println("Stack Underflow");
        }else{
            num = a[top];
            top = top-1;
        }

        return num;
    }

    public void display(){
        for(int i=top;i>-1;i--){
            System.out.println(a[i]);
        }
    }


}


public class stack{
    public static void main(String[] args) {
        Stack s1 = new Stack(10);

        s1.push(1);
        s1.push(2);
        s1.push(3);
        s1.push(4);
        s1.push(5);
        s1.push(6);
        s1.push(7);
        s1.push(8);
        s1.push(9);
         s1.push(10);

         s1.display();
        // s1.push(11);
        System.out.println("Popped Element : "+ s1.pop());
        System.out.println("Popped Element : "+ s1.pop());
        System.out.println("Popped Element : "+ s1.pop());
        System.out.println("Popped Element : "+ s1.pop());
        System.out.println("Popped Element : "+ s1.pop());
        System.out.println("Popped Element : "+ s1.pop());
        System.out.println("Popped Element : "+ s1.pop());
        System.out.println("Popped Element : "+ s1.pop());
        System.out.println("Popped Element : "+ s1.pop());
        System.out.println("Popped Element : "+ s1.pop());
        System.out.println("Popped Element : "+ s1.pop());

        System.out.println("New List : ");

        s1.display();

    }
}